### PR TITLE
Fix for disabled button can be pressed issue

### LINKED ISSUE #6957: Disabled button can be pressed
**Describe the bug**
Button stays focused when IsEnabled set to false and can process keyboard ENTER event.

**To Reproduce**
Run this window and follow the screen instructions:
```
 public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
#if DEBUG
            this.AttachDevTools();
#endif
        }

        private void InitializeComponent()
        {
            AvaloniaXamlLoader.Load(this);

            var button = new Button
            {
                Content = "Tabulate to this button"
            };

            Content = button;
            button.GotFocus += (_, _) =>
            {
                button.IsEnabled = false;
                button.Content =
                    $"Now this button is disabled ({nameof(IsEnabled)}:{button.IsEnabled},{nameof(IsEffectivelyEnabled)}:{button.IsEffectivelyEnabled}), but you can still press Enter";
                button.Click += (_, _) =>
                {
                    button.Content = "It just has been clicked.";
                };
            };
        }
    }
```

**Expected behavior**
1) Probably focus must be moved away from the control once is has been disabled
2) Probably a control must not process input events once is has been disabled

**Screenshots**
image

**Desktop (please complete the following information):**

- OS: Linux
- Version 0.10.10

### PR DESCRIPTION
**Pull request by Abdesol**
## What does the pull request do?
Adds IsEnabled and IsEffectivelyEnabled condition to OnGotFocus(from InputElement) and OnClick(from Button)

## What is the updated/expected behavior with this PR?
If a controller is disabled(IsEnabled set to false), The GotFocus and Click event won't fire(work) unless the button is enabled.

## Fixed issues
Fixes https://github.com/AvaloniaUI/Avalonia/issues/6957

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=7205) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=7205) before we can accept your contribution.<br/><br/>:x: Abdesol [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=7205)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=7205) it.</sub>

File: src/Avalonia.Controls/Button.cs [hunk 1]
```diff
@@ -304,15 +304,18 @@ protected override void OnKeyUp(KeyEventArgs e)
...
-            if (!e.Handled && Command?.CanExecute(CommandParameter) == true)
-            {
-                Command.Execute(CommandParameter);
-                e.Handled = true;
+                if (!e.Handled && Command?.CanExecute(CommandParameter) == true)
+                {
+                    Command.Execute(CommandParameter);
+                    e.Handled = true;
+                }
             }
```
Comment (maxkatz6) [inline src/Avalonia.Controls/Button.cs:319]: Can you add tests for that? By raising key down event on the button while it's disabled (and another test when parent is disabled).
Also, I think it should be enough to use IsEffectivelyEnabled.

File: src/Avalonia.Input/InputElement.cs [hunk 2]
```diff
@@ -516,9 +516,12 @@ protected override void OnAttachedToVisualTreeCore(VisualTreeAttachmentEventArgs
...
         {
-            var isFocused = e.Source == this;
-            _isFocusVisible = isFocused && (e.NavigationMethod == NavigationMethod.Directional || e.NavigationMethod == NavigationMethod.Tab);
-            IsFocused = isFocused;
+            if (IsEnabled & IsEffectivelyEnabled)
+            {
+                var isFocused = e.Source == this;
+                _isFocusVisible = isFocused && (e.NavigationMethod == NavigationMethod.Directional || e.NavigationMethod == NavigationMethod.Tab);
+                IsFocused = isFocused;
+            }
```
Comment (maxkatz6) [inline src/Avalonia.Input/InputElement.cs:524]: Don't think it's correct change.
As I understand disabled elements still get focused (I haven't tested it to confirm), which means with this change old focused element will be unfocused, new element will be focused internally, but only IsFocused property won't be updated.
Probably you need to take a look at FocusManager class, there should be logic to decide if element should be focused.

Comment (maxkatz6) [reply src/Avalonia.Input/InputElement.cs:524]: Or actually, it shouldn't get focus when it's disabled. #6957 problem is about disabling element **while** it is focused. So, this changes block is not needed.

We might want to unfocused element when it gets disabled, but for this PR I think Button.cs changes are enough. Also it gets complicated like what exactly should be focused instead, should be compared with WPF/UWP.

File: src/Avalonia.Controls/Button.cs (hunk 1 above)
Comment (Abdesol) [reply src/Avalonia.Controls/Button.cs]: I accidentally closed this pr while reverting the commits.
https://github.com/AvaloniaUI/Avalonia/pull/7206 This is the new pr

**Pull request closed without merging**